### PR TITLE
dvi

### PR DESCRIPTION
**Pull request by burrbull**
@Mrmaxmeier  Can you test this also?

### CODE REVIEW

**Comment by Mrmaxmeier** [comment]
After e7e7126:
```
total regressed (reg): 53
total regressed (fix): 84
    fixed regressions: 0
      new regressions: 31
changed but not fixed: 3
```

Here's the changes/regressions:
https://tt.ente.ninja/#/compare/oxidize-v0.1.11-725-g83fc2fa8/dvi2-v0.1.11-745-ge7e7126c

**Comment by burrbull** [comment]
1702.06424
p.5 FIG. 3: Absent
Looks like a bug when insert external pdf.

**Comment by burrbull** [comment]
I'he retried 1702.06424 locally and FIG. 3 in place, but PDF 13 bytes longer.
image

**Comment by burrbull** [comment]
@Mrmaxmeier can you write small howto for your testing system?

**Comment by burrbull** [comment]
Sometimes
```
warning: Could not read object from object stream.
warning: Cannot read PDF document catalog. Broken PDF file?
warning: pdf: image inclusion failed for "fig3.pdf".
warning: Could not find image resource...
warning: Interpreting special command image (pdf:) failed.
warning: >> at page="5" position="(323.1411065924658, 577.5958898028875)" (in PDF)
warning: >> xxx "pdf:image matrix 1.0 0.0 0.0 1.0 -50.80945 -427.39725 page 0 pa"
warning: Object @figure.3 already defined.
```
instead of just
```
warning: Object @figure.3 already defined.
```

**Comment by burrbull** [comment]
Split this PR on #149 #150 #151

**Comment by burrbull** [comment]
@Mrmaxmeier can you retest again?

**Comment by burrbull** [comment]
Left only 1 commit. Others in #148

**Comment by Mrmaxmeier** [comment]
I'm currently looking into GitHub's API for CI services. I'd be nice to have this as automated feedback on every pull request.

Here's another manual run:
https://tt.ente.ninja/#/compare/stopgap-for-486-v0.1.11-761-gc9452cf9/dvi2-v0.1.11-762-geba14c6a

> 1702.06424
> p.5 FIG. 3: Absent
> Looks like a bug when insert external pdf.

Looks like (only) this issue is still present.

It might be an nondeterministic issue that was introduced in some earlier commit though. We'll have to investigate...

**Comment by cormacrelf** [comment]
(You could emulate crater’s comment-based usage with https://github.com/machine-learning-apps/actions-chatops/blob/master/README.md )

File: dpx/src/specials/html.rs [hunk 1]
```diff
@@ -935,24 +935,16 @@ pub unsafe extern "C" fn spc_html_at_end_document() -> i32 {
...
-            p as *const libc::c_void,
-            b"html:\x00" as *const u8 as *const i8 as *const libc::c_void,
-            strlen(b"html:\x00" as *const u8 as *const i8),
-        ) == 0
-    {
-        return true;
+pub fn spc_html_check_special(buf: &[u8]) -> bool {
+    let mut i = 0;
+    for &p in buf {
+        if crate::isblank(p as _) == 0 {
```
Comment (cormacrelf) [inline dpx/src/specials/html.rs]: Should this be isspace? isblank only checks space and tab.

**Comment by cormacrelf** [comment]
Seems more likely this was from #150 which actually decodes catalogs.

**Comment by burrbull** [comment]
> Seems more likely this was from #150 which actually decodes catalogs.

What are you talking about?

File: dpx/src/specials/html.rs (hunk 1 above)
Comment (burrbull) [reply dpx/src/specials/html.rs]: Looks like copy-paste typo. Good catch.

**Comment by burrbull** [comment]
> Looks like (only) this issue is still present.
> 
> It might be an nondeterministic issue that was introduced in some earlier commit though. We'll have to investigate...

I thought I've fixed this in #151 .

**Comment by burrbull** [comment]
If that's what I'm thinking:
https://github.com/tectonic-typesetting/tectonic/commit/c9452cf93006b2513849ecb6af54b755db907276
it can be added under cargo `source-time` feature or in `build.rs`.

**Comment by cormacrelf** [comment]
>> Seems more likely this was from #150 which actually decodes catalogs. ...

@burrbull the failure includes “Cannot read PDF document catalog” doesn’t it?

**Comment by burrbull** [comment]
Does anyone mind to merge this?

**Comment by Mrmaxmeier** [comment]
> It might be an nondeterministic issue that was introduced in some earlier commit though. We'll have to investigate...

I have not found the culprit yet but this seems like it was introduced in an earlier commit.

**Pull request merged**